### ISSUE TITLE
data_collator_wrapper

### ISSUE THREAD

**Issue by gyangfan** [alibaba/ROLL#184]
def data_collator_wrapper(data_collator):
    @functools.wraps(data_collator)
    def wrapper(features: Sequence[Dict[str, Any]]):
        labels_key = [k for k in features[0].keys() if k.endswith("labels")]
        input_ids_key = [k for k in features[0].keys() if k.endswith("input_ids")]
        for feature in features:
            if len(labels_key) == 0:  # pt
                feature["labels"] = deepcopy(feature["input_ids"])[1:]
            for k in labels_key:
                feature[k] = feature[k][1:]
            for k in input_ids_key:
                feature[k] = feature[k][:-1]
            for k in ["attention_mask", "position_ids"]:
                if k in feature:
                    feature[k] = feature[k][1:]
        return data_collator(features)

    return wrapper

attention_mask 和 position_ids 是不是应该是 [:-1]

**Comment by liu-zichen** [comment]
Yes, we will fix it soon.

**Comment by PanAndy** [comment]
https://github.com/alibaba/ROLL/issues/207
